src/librustc_trans/trans/base.rs

```diff
@@ -1616,6 +1616,7 @@ pub fn new_fn_ctxt<'a, 'tcx>(ccx: &'a CrateContext<'a, 'tcx>,
         param_substs: param_substs,
         span: sp,
         block_arena: block_arena,
+        lpad_arena: TypedArena::new(),
```

Comment:
Ah, I see it, you’re still storing a pointer inside a `Block`. A few thoughts:

1. Size of `BlockS` does not really matter, because we only ever pass a `Block` around, which is a `pub type Block<'blk, 'tcx> = &'blk BlockS<'blk, 'tcx>;` – i.e. is a reference and there’s barely any performance overhead passing a pointer to larger or smaller struct;
2. From my experimentation with implementing MSVC LPs in MIR, all that really needs to be stored around for MIR itself is a single ValueRef (a pointer, basically) inside MirContext. All the landing-pad related things in `Block` are for the old trans to use, and I wouldn’t bother much changing it (but since you’ve already changed it, its fine, I guess?).

I’d still like it split up into a separate commit so it is easier to review and these changes do not mix with the Builder refactor.